test_scripts/Defects/4_5/1880_DefaultAndRPCTimeoutGENERIC_ERROR.lua

```diff
@@ -0,0 +1,403 @@
+  else
+    -- timeout default value from mobile API
+    SliderDuration = 10000
+  end
+  local RespTimeout = DefaultTimeout + SliderDuration
+  local RequestTime
+  local RespTime
+  local TimeBetweenReqRes
+  local cid = self.mobileSession1:SendRPC("Slider", params)
+  RequestTime = timestamp()
+  EXPECT_HMICALL("UI.Slider")
+  self.mobileSession1:ExpectResponse(cid, {success = false, resultCode = "GENERIC_ERROR"})
+  :Timeout(RespTimeout + 1000)
+  :ValidIf(function()
+      RespTime = timestamp()
+      TimeBetweenReqRes = RespTime - RequestTime
+      if
+      TimeBetweenReqRes > RespTimeout - 1000 and
+      TimeBetweenReqRes < RespTimeout + 1000 then
+        return true
+      else
+        return false, "SDL triggers timeout earlier then expected(".. tostring(RespTimeout) .." sec), after " ..
+        tostring(TimeBetweenReqRes) .. " sec. \n SDL must use Slider timeout + default timeout."
+      end
+    end)
+end
+
+local function ScrollableMessage(params, self)
+  local ScrMesDuration
+  if params.timeout then
+    ScrMesDuration = params.timeout
+  else
+    -- timeout default value from mobile API
+    ScrMesDuration = 30000
+  end
+  local RespTimeout = DefaultTimeout + ScrMesDuration
+  local RequestTime
+  local RespTime
+  local TimeBetweenReqRes
+  local cid = self.mobileSession1:SendRPC("ScrollableMessage", params)
+  RequestTime = timestamp()
+  EXPECT_HMICALL("UI.ScrollableMessage")
+  self.mobileSession1:ExpectResponse(cid, {success = false, resultCode = "GENERIC_ERROR"})
+  :Timeout(RespTimeout + 1000)
+  :ValidIf(function()
+      RespTime = timestamp()
+      TimeBetweenReqRes = RespTime - RequestTime
+      if
+      TimeBetweenReqRes > RespTimeout - 1000 and
+      TimeBetweenReqRes < RespTimeout + 1000 then
+        return true
+      else
+        return false, "SDL triggers timeout earlier then expected(".. tostring(RespTimeout) .." sec), after " ..
+        tostring(TimeBetweenReqRes) .. " sec. \n SDL must use ScrollableMessage timeout + default timeout."
+      end
+    end)
+end
+
+local function CreateInteractionChoiceSet(self)
+  local cid = self.mobileSession1:SendRPC("CreateInteractionChoiceSet", {
+      interactionChoiceSetID = 100,
+      choiceSet = {
+        {
+          choiceID = 100,
+          menuName ="Choice100",
+          vrCommands = {
+            "Choice100",
+          }
+        }
+      }
+    })
+  EXPECT_HMICALL("VR.AddCommand")
+  :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+  self.mobileSession1:ExpectResponse(cid, {success = true, resultCode = "SUCCESS"})
+end
+
+local function PerformInteraction(params, self)
+  local PIDuration
+  local RequestTime
+  local RespTime
+  local TimeBetweenReqRes
+  local TimeToResponseForVR = 2000
+  local RespTimeout
+  if params.timeout then
+    PIDuration = params.timeout
+    RespTimeout = DefaultTimeout + 2*PIDuration
+  else
+    PIDuration = 10000
```

Comment:
@GetmanetsIrina Propose to read such values form API.XML file instead of hardcoding

Reply:
@AByzhynar From my point of view it would be better to use hard coded value in this case. Reasons:
 - this is just a one script for one defect
 - just only one value that can be read from API

Reply:
@AByzhynar Agreed

Reply:
@AByzhynar Done in 9fa904e